packages/app/src/components/DBRowTable.tsx

```diff
@@ -681,24 +682,30 @@ export const RawLogTable = memo(
   },
 );
 
-function mergeSelectWithPrimaryKey(
+function mergeSelectWithPrimaryAndPartitionKey(
   select: SelectList,
   primaryKeys: string,
+  partitionKey: string,
 ): { select: SelectList; additionalKeysLength: number } {
-  const keys = primaryKeys.split(',').map(k => k.trim());
+  const partitionKeyArr = partitionKey
+    .split(',')
+    .map(k => extractColumnReference(k.trim()))
+    .filter((k): k is string => k != null && k.length > 0);
```

Comment:
curious why do we need to extract column reference here? is is because we try to reduce the redundant select?

Reply:
partition keys can look like `toDate(Start)`, we really just want `Start` in that case in the select statement so that when we feed it into the where statement it'll correctly do `Start = ...` and allow the partition key to select the correct partitions.

Reply:
ah I totally forgot about this. I now recall we have queries like this in v1 (mostly metrics related). I guess what is interesting is the column comparison is slightly more optimized than the original expression (ex: `toDate(Start) = ...`)